src/photon/photon_scheduler.c

```diff
@@ -346,14 +364,40 @@ int heartbeat_handler(event_loop *loop, timer_id id, void *context) {
   return LOCAL_SCHEDULER_HEARTBEAT_TIMEOUT_MILLISECONDS;
 }
 
+void start_new_worker(local_scheduler_state *state) {
+  /* We can't start a worker if we don't have the path to the worker script. */
+  CHECK(state->worker_path != NULL);
+  /* Format a command to start a worker. */
+  UT_string *start_worker_command;
+  utstring_new(start_worker_command);
+  utstring_printf(start_worker_command,
+                  "python %s "
+                  "--node-ip-address=%s "
+                  "--object-store-name=%s "
+                  "--object-store-manager-name=%s "
+                  "--local-scheduler-name=%s "
+                  "--redis-address=%s:%d",
+                  state->worker_path, state->node_ip_address,
+                  state->plasma_store_socket_name,
+                  state->plasma_manager_socket_name,
+                  state->local_scheduler_socket_name, state->redis_address,
+                  state->redis_port);
+  /* Launch the process to create the worker. */
+  FILE *p = popen(utstring_body(start_worker_command), "r");
```

Comment:
don't we want to keep track of all the started workers? I think we should keep a data structure with worker handles in the LS state.